### ISSUE TITLE
N64 build error on iOS (ld: framework not found PVMupen64PlusVideoGlideN64)

### ISSUE THREAD

**Issue by OutRunner** [Provenance-Emu/Provenance#764]
Hi Guys,

I don't know if you want reports of issues on the develop branch so apologies if this is unwanted. I'm getting the below error when trying to build the latest iOS version on both the debug and release schemes. It's builds fine on tvOS (Mario Kart 64 is running through the attract mode at the moment!) so presumably it's a minor config issue on the iOS version (unless I've done something wrong). 

I tried the usual stuff like checking the framework was included on the Build Phases tab, cleared the DerivedData directory, made sure there weren't any spaces in the paths used, deleted and re-downloaded the repo, restarted Xcode and even rebooted my MacBook but the error would always be there after building. 

Anyway, I hope this is helpful, let me know if you want me to try anything.

Regards,
Si

```
Ld /Users/myusername/Library/Developer/Xcode/DerivedData/Provenance-fldutmdeoynkohgvbzjcsljprsto/Build/Intermediates.noindex/Provenance.build/Release-iphoneos/Provenance.build/Objects-normal/armv7/Provenance normal armv7
    cd /Users/myusername/Desktop/Provenance
    export IPHONEOS_DEPLOYMENT_TARGET=8.0
    export PATH="/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/usr/bin:/Applications/Xcode.app/Contents/Developer/usr/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin"
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang++ -arch armv7 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS11.2.sdk -L/Users/myusername/Library/Developer/Xcode/DerivedData/Provenance-fldutmdeoynkohgvbzjcsljprsto/Build/Products/Release-iphoneos -L/Users/myusername/Library/Developer/Xcode/DerivedData/Provenance-fldutmdeoynkohgvbzjcsljprsto/Build/Products -F/Users/myusername/Library/Developer/Xcode/DerivedData/Provenance-fldutmdeoynkohgvbzjcsljprsto/Build/Products/Release-iphoneos -F/Users/myusername/Desktop/Provenance -F/Users/myusername/Desktop/Provenance/Realm-iOS -filelist /Users/myusername/Library/Developer/Xcode/DerivedData/Provenance-fldutmdeoynkohgvbzjcsljprsto/Build/Intermediates.noindex/Provenance.build/Release-iphoneos/Provenance.build/Objects-normal/armv7/Provenance.LinkFileList -Xlinker -rpath -Xlinker @executable_path/Frameworks -miphoneos-version-min=8.0 -dead_strip -Xlinker -object_path_lto -Xlinker /Users/myusername/Library/Developer/Xcode/DerivedData/Provenance-fldutmdeoynkohgvbzjcsljprsto/Build/Intermediates.noindex/Provenance.build/Release-iphoneos/Provenance.build/Objects-normal/armv7/Provenance_lto.o -fobjc-arc -fobjc-link-runtime -stdlib=libc++ -L/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/lib/swift/iphoneos -Xlinker -add_ast_path -Xlinker /Users/myusername/Library/Developer/Xcode/DerivedData/Provenance-fldutmdeoynkohgvbzjcsljprsto/Build/Intermediates.noindex/Provenance.build/Release-iphoneos/Provenance.build/Objects-normal/armv7/Provenance.swiftmodule -ObjC -framework PVSupport -framework PicoDrive -framework RealmSwift -framework ProSystem -weak_framework PVRSPCXD4 -weak_framework PVMupen64Plus -framework PVMupen64PlusVideoGlideN64 -lxml2 -framework PVGB -lz -lstdc++ -lc++ -lsqlite3 -framework PVGenesis -framework PVAtari800 -framework AVFoundation -framework PVPokeMini -framework AssetsLibrary -framework CFNetwork -framework MobileCoreServices -framework Security -framework SystemConfiguration -framework CoreData -framework AudioToolbox -framework GLKit -framework PVGBA -weak_framework PVMupen64PlusVideoRice -framework OpenGLES -weak_framework PVMupen64PlusRspHLE -framework PVSNES -framework UIKit -framework Foundation -framework PVStella -framework CoreGraphics -framework PVFCEU -framework PVMednafen -framework Realm -Xlinker -dependency_info -Xlinker /Users/myusername/Library/Developer/Xcode/DerivedData/Provenance-fldutmdeoynkohgvbzjcsljprsto/Build/Intermediates.noindex/Provenance.build/Release-iphoneos/Provenance.build/Objects-normal/armv7/Provenance_dependency_info.dat -o /Users/myusername/Library/Developer/Xcode/DerivedData/Provenance-fldutmdeoynkohgvbzjcsljprsto/Build/Intermediates.noindex/Provenance.build/Release-iphoneos/Provenance.build/Objects-normal/armv7/Provenance

ld: framework not found PVMupen64PlusVideoGlideN64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

**Comment by Cepheidsong** [comment]
I'm not at home to confirm but I'm pretty sure this is the fix for you:  You have to codesign Spotlight as well, which is for the ios side.  In xcode you'll see like on the left, Provenance, ProvenanceTV, TopShelf, and Spotlight.  Put in your info that you put in Provenance to sign in Spotlight too and it should work.

**Comment by OutRunner** [comment]
I think Spotlight is correctly signed, does this look OK?

<img width="1680" alt="xcode" src="https://user-images.githubusercontent.com/2256524/37867193-915637fe-2f8c-11e8-8429-109ba18fe24a.png">

**Comment by Cepheidsong** [comment]
Yeah looks okay.  When I tried a build awhile ago with the same issue going on, that fixed it for me, so give that a run and see if it builds ok- but at the top where it says Generic IOS Device don't forget to pick your plugged in phone on the list.

**Comment by jasarien** [comment]
I've just made a new commit to `develop` to fix this. Can you pull the latest `develop` and try again?

**Comment by Cepheidsong** [comment]
(just so you know I have the previous develop build on my phone right now so it did build right in my case)

**Comment by OutRunner** [comment]
@jasarien I’ve pulled the latest `develop` and that has fixed the PVMupen64PlusVideoGlideN64 issue, however it’s now giving the same error on PVMupen64PlusRspHLE. I’m sorry I haven’t fixed it myself, I can see what’s changed with the schemes from your commit but I’m not sure where in Xcode to add another one. I tried adding a new scheme under the PVMupen64Plus project but that ended up under the main build targets so I’m clearly doing it wrong.

**Comment by jasarien** [comment]
Don’t create a new scheme. Edit the existing ones for `Provenance` and `Provenance-Release`. In the build section, add the missing framework.

**Comment by jasarien** [comment]
Ok, so I updated all the targets to include any missing frameworks from their build lists. Pull from develop and try again please?

**Comment by OutRunner** [comment]
I was just about to say I added the missing framework and got it working. ;) I've stashed and pulled your fix and that's working too, thanks for looking at it.

**Comment by jasarien** [comment]
Fixed in 217a40beba04e91e9840543bdca8afbcf85bcd14
